### PR TITLE
Docs and API Reference generation with docfx

### PR DESCRIPTION
**Pull request by markvincze**
This PR adds documentation and API Reference generation using [docfx](https://dotnet.github.io/docfx/).
The way it works is that a new `docs` folder contains all the config and setup. The docs can be generated with the `docfx` CLI, which puts the output (by default) to `docs/_site`.
The idea is to host the docs with GitHub pages in the same repo in the `gh-pages` branch. The added `releaseDocs.sh` script self-contained, it generates the docs, checks out the repo, adds the new docs and pushes them to the `gh-pages` branch. For now I've only been executing this script in my machine. If we want, in the future we can set this up so this runs on AppVeyor automatically.
Ignore everything in the `docs/_exported_templates` folder, that's just the default template `docfx` is using. I added it to the repo, because I wanted to customize the navigation bar a bit.
What I did with the docs is that I copied everything from the `README`  to the docs, and split into a couple of pages. I also adjusted the introduction docs a bit, you can find it in `docs/documentation/intro.md` (or on the site at https://travix-international.github.io/Hystrix.Dotnet/documentation/intro.html).
I didn't change the `README` yet, I think in a followup PR we can make it shorter, and add a link saying "details in the documentation".
I pushed the docs already to the `gh-pages` branch, so you can take a look at it here: https://travix-international.github.io/Hystrix.Dotnet/index.html

Addresses #21.

### CODE REVIEW

**Comment by 1kastner** [comment]
That looks nice!

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 A combination of circuit breaker and timeout. The .NET version of the open source [Hystrix library](https://github.com/Netflix/Hystrix) built by Netflix.
 
-[![Build Status .NET](https://ci.appveyor.com/api/projects/status/github/Travix-International/Hystrix.Dotnet?svg=true)](https://ci.appveyor.com/project/Travix-International/Hystrix.Dotnet/)
+[![Build Status .NET](https://ci.appveyor.com/api/projects/status/1a09t29hcvfo1bbq/branch/master?svg=true)](https://ci.appveyor.com/project/Travix-International/Hystrix.Dotnet/)
```
Comment (JorritSalverda) [inline README.md:5]: Did appveyor change their status urls?

**Review by JorritSalverda — APPROVED** [review]

**Comment by markvincze** [comment]
I made some small changes, and I made the `README` shorter and pointing to the Documentation.  
@JorritSalverda can you take a last look here if the new shorter `README` looks good? https://github.com/Travix-International/Hystrix.Dotnet/tree/docfx-generation

File: README.md (hunk 1 above)
Comment (markvincze) [reply README.md:5]: We were using a wrong link, which would show the status of the latest build *regardless of the branch*. So if a build in a PR branch failed, that made the badge red on the home page.

Although I don't know why the structure of the badge route is different (it looks kinda obfuscated now). I just went to the AppVeyor "Badges" page and grabbed the link from there. Maybe they support both, and now this is the default.

**Pull request merged**